### ISSUE TITLE
Clean up temp directory after CREATE or INSERT in Hive

### ISSUE THREAD

**Issue by dain** [prestodb/presto#3731]
After the CREATE or INSERT for Hive finished, the temp directory will have some left over empty directories.  For example, I ran 4 insert commands and was left with:
```
/tmp/presto-dain/22041385-5745-4961-ae03-6cca9b6640ff/ship_priority=0
/tmp/presto-dain/3b29ca9b-0885-45bc-8d65-dcff53274c02/ship_priority=0
/tmp/presto-dain/c81fb4a3-91fd-41d6-8421-1aca6add7501/ship_priority=0
/tmp/presto-dain/e529708f-e9a0-4ba0-b499-a1ff28f35ef9/ship_priority=0
```

The cleanup code in the `HiveMetadata` should attempt to remove the empty temp directories after the command complete (either successfully or by failure).

**Comment by kokosing** [comment]
I can take this one.

**Comment by kokosing** [comment]
Pull request to fix this: https://github.com/facebook/presto/pull/3851

**Comment by yuananf** [comment]
fixed by #5352
